### DISCUSSION TITLE
How to handle specified styles in the unocss.config.ts file

### DISCUSSION THREAD

**Discussion by aliuq** [unocss/unocss#2164] [Q&A]
Like the example below

```ts
export default defineConfig({
  theme: {
    colors: {
      main: 'colors.blue.500' // text-blue-5
    }
  }
})
```

**Comment by zyyv** [comment]
Can you give an example of your usage?

**Reply by aliuq** [reply]
[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAuBTJmDKmATmAHYDm6AfAFBRTAAOsiKGOAHpgLQzak4iUTj0YkAtgiIBPaiGziA9sAD0jWqvARaQA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHFYALaDGb0YrAFxwA2itVHaMVKxgBlGFGxFaAGjgADVqjGpmMaAFojADpLawBdfQUjf38QbH8Ad2BkGHV9ACYABjBeMOMY5ggxfyJhMAzs3LMCgkssGFj4pJS05mExZgUAN2EoBRi4rETk1IBKAH1-cfTxgCpxnr6a4tKwUdGw0bcw0zDnU1TUEFRDZVVVQrFoA0RTc7gwBxBegE9DWkvroIAjMXpUIIAVkymS491IpghpFGQA&options=N4IgzgLgTglgxhEAuaBXApgGhNAhgOzADMB7KAW2SNwBsx0BfIA)

See code in playground, Should be useful when building a theme

**Reply by zyyv** [reply]
How about use `shortcuts`?
[Playground](https://uno.antfu.me/play/?html=DwEwlgbgBAxgNgQwM5ILwCIAuBTJmDKmATmAHYDm6AfAFBRTAAOsiKGOAHpgLQzak4iUTj0YkAtgiIBPYdi7cxYSTO4AmAAwbqIbOID2wAPSNax8BFpA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQnqAHpLBQYAhvQA28NJhz5CwIgAoEnOHFYALaDGb0YrAFxwA2itXGA9AD0YfGAFowUUMKgBPBQH59dhQDoA1ACUgR4AJOYANHAKRlHIALqBcAC8AHxwAAY2vPYARmL0qKEIyHAemXbFyKQZcIa0tDXxEaaqJmZmtDasMADKME5YRLQtHaoZrKhiqMww0HZGvt0w8foxdnYg2HYA7sDIMOr6AEwADGC88cYbzBBidkTCYCfnl203BDZY9ltYu-uHfTMYRiZgKABuLgUG1%2B-wO6kCAH07IjjoiAFSIyFQaF2W73R5gYKJDKjMzNUwU1SHVAgVCGZQdfHQAyIVpmRzONz1ZlQVi%2BfKFXwAVlOpxG7NIZKlnFIgU4QA&options=N4IgzgLgTglgxhEAuaBXApgGhNAhgOzADMB7KAW2SNwBsx0BfIA)

**Reply by aliuq** [reply]
Not only used by `text-`, also should be used by border or background

**Reply by zyyv** [reply]
You should custom your theme colors.

```ts
import { colors } from '@unocss/preset-mini'

export default defineConfig({
  theme: {
    colors: {
      primary: colors.blue['500']
    }
  }
})
```

Then, you can use `text-primary` `bg-primary`

**Reply by aliuq** [reply]
It works, but should typed `colors.blue['500']`, it seems not friendly
